src/Common/tests/Collections/IEnumerableTest.cs

```diff
@@ -0,0 +1,428 @@
+
+            if (!IsResetNotSupported)
+            {
+                Assert.Throws<InvalidOperationException>(
+                    () => enumerator.Reset());
+            }
+        }
+
+        private void VerifyEnumerator(
+            object[] expectedItems,
+            IEnumerator enumerator)
+        {
+            VerifyEnumerator(expectedItems, enumerator, 0, expectedItems.Length, true, true);
+        }
+
+        private void VerifyEnumerator(
+            object[] expectedItems,
+            IEnumerator enumerator, int startIndex, int count, bool validateEnd, bool validateStart)
+        {
+            bool needToMatchAllExpectedItems = count - startIndex
+                                               == expectedItems
+                                                      .Length;
+            if (validateStart)
+            {
+                for (int i = 0; i < 3; i++)
+                {
+                    Assert.Throws<InvalidOperationException>(
+                        () => enumerator.Current);
+                }
+            }
+
+
+            int iterations;
+            if (CollectionOrder == CollectionOrder.Unspecified)
+            {
+                BitArray itemsVisited = new BitArray(needToMatchAllExpectedItems ? count : expectedItems.Length, false);
+                for (iterations = 0;
+                     iterations < count && enumerator.MoveNext();
+                     iterations++)
+                {
+                    var currentItem = enumerator.Current;
+                    var itemFound = false;
+                    for (int i = 0; i < itemsVisited.Length; ++i)
+                    {
+                        if (!itemsVisited[i]
+                            && Equals(currentItem, expectedItems[i + (needToMatchAllExpectedItems ? startIndex : 0)]))
+                        {
+                            itemsVisited[i] = true;
+                            itemFound = true;
+                            break;
+                        }
+                    }
+                    Assert.True(itemFound, "itemFound");
+
+                    for (int i = 0; i < 3; i++)
+                    {
+                        var tempItem = enumerator.Current;
+                        Assert.Equal(currentItem, tempItem);
+                    }
+                }
+                if (needToMatchAllExpectedItems)
+                {
+                    for (int i = 0; i < itemsVisited.Length; i++)
+                    {
+                        Assert.True(itemsVisited[i]);
+                    }
+                }
+                else
+                {
+                    var visitedItemCount = 0;
+                    for (int i = 0; i < itemsVisited.Length; i++)
+                    {
+                        if (itemsVisited[i])
+                            ++visitedItemCount;
+                    }
+                    Assert.Equal(count, visitedItemCount);
+                }
+            }
+            else
```

Comment:
You have an if on an enum match, then just an else.  If it's just "if/else", then it should be a boolean.  If you expect future expansion, then if == a, else if == b, else throw.